pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeConvert.java

```diff
@@ -35,7 +36,7 @@ public class DateTimeConvert {
   private DateTimeGranularitySpec _granularitySpec;
 
   @ScalarFunction(names = {"dateTimeConvert", "date_time_convert"})
-  public String dateTimeConvert(String timeValueStr, String inputFormatStr, String outputFormatStr,
+  public Object dateTimeConvert(String timeValueStr, String inputFormatStr, String outputFormatStr,
```

Comment:
this is not the right fix. we need to register it as SqlOperator b/c the return type varies based on input CC @Jackie-Jiang

Reply:
there is a type inference here: https://github.com/apache/pinot/blob/master/pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java#L125

Reply:
The response data type is correct, please check.